Google.Api.CommonProtos/Rpc/StatusPartial.cs

```diff
@@ -0,0 +1,132 @@
+            {
+                return null;
+            }
+            return any.Unpack<T>();
+        }
+
+        /// <summary>
+        /// Iterate over all the messages in the <see cref="Google.Rpc.Status.Details"/>
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Iterate over the messages in the <see cref="Google.Rpc.Status.Details"/> that are messages
+        /// in the <see href="https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto">
+        /// standard set of error types</see> defined in the richer error model. Any other messages found in
+        /// the Details are ignored and not returned.
+        /// </para>
+        /// <para>
+        /// <example>
+        /// Example:
+        /// <code>
+        /// foreach (var msg in status.UnpackDetailMessages()) {
+        ///   switch (msg) {
+        ///     case ErrorInfo errorInfo:
+        ///          // Handle errorInfo ...
+        ///          break;
+        ///     // Other cases ...
+        ///   }
+        /// }
+        /// </code>
+        /// </example>
+        /// </para>
+        /// </remarks>
+        /// <returns></returns>
+        public IEnumerable<IMessage> UnpackDetailMessages()
+        {
+            return UnpackDetailMessages(StandardErrorTypeRegistry.Registry);
+        }
+
+        /// <summary>
+        /// Iterate over all the messages in the <see cref="Google.Rpc.Status.Details"/> that match types
+        /// in the given <see cref="TypeRegistry"/>
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Iterate over the messages in the <see cref="Google.Rpc.Status.Details"/> that are messages
+        /// in the given <see cref="TypeRegistry"/>. Any other messages found in the Details are ignored
+        /// and not returned.  This allows iterating over custom messages if you are not using the
+        /// standard set of error types defined in the rich error model.
+        /// </para>
+        /// <para>
+        /// <example>
+        /// Example:
+        /// <code>
+        /// TypeRegistry myTypes = TypeRegistry.FromMessages(
+        ///   new MessageDescriptor[] {
+        ///     FooMessage.Descriptor, BarMessage.Descriptor
+        ///   });
+        ///   
+        /// foreach (var msg in status.UnpackDetailMessages(myTypes)) {
+        ///   switch (msg) {
+        ///     case FooMessage foo:
+        ///          // Handle foo ...
+        ///          break;
+        ///     // Other cases ...
+        ///   }
+        /// }
+        /// </code>
+        /// </example>
+        /// </para>
+        /// </remarks>
+        /// <param name="registry"></param>
+        /// <returns></returns>
+        public IEnumerable<IMessage> UnpackDetailMessages(TypeRegistry registry)
+        {
+            foreach (var any in Details)
+            {
+                var msg = any.Unpack(registry);
+                if (!(msg is null))
```

Comment:
Perhaps just use an expression bodied member of `Details.Select(any => any.Unpack(registry)).OfType<IMessage>());`?